src/core/lib/event_engine/posix_engine/posix_engine.cc

```diff
@@ -92,6 +95,18 @@ void PosixEventEngine::Run(EventEngine::Closure* closure) {
   executor_.Run(closure);
 }
 
+absl::Status PosixEventEngine::WaitForPendingTasks(Duration timeout) {
```

Comment:
What actions would one take if the timeout expired?

Reply:
In benchmark and testing cases, I'd expect we assert a failure. If some production code is using it to attempt a safe engine shutdown, maybe logging to indicate a leak, similar to what we have on gRPC shutdown today.

Reply:
It may be worth going for a `WaitForPendingTasksOrDie(timeout)` API then - it doesn't seem like it'd be safe to proceed if this fails.

Reply:
That could work. The one exception I can imagine at the moment is an engine that wants to offer a best effort flush of its queue on shutdown or fork, but that can be managed internally to the engine.

Some of the tests in the test suite will need to be death tests now, which are not terribly friendly on all platforms. Do you think we should make the change anyway?

Reply:
The best effort flush thing is a really dangerous pattern. It's never worked well for iomgr, I should not have added it, and we should not bring it forward. It's a relic pattern from when we didn't have sanitizers, but since it leaves the application in an undefined state it leaves the application vulnerable to data corruption or other problems post call.

Which is to say, either the application relies on callbacks being done (meaning this should be an OrDie function) or applications do not rely on being able to do this and so we should not export this function at all.

Reply:
Changed to `WaitForPendingTasksOrDie`, with one death test.